core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -1035,7 +1035,11 @@ object AutoTuner extends Logging {
     representer.getPropertyUtils.setSkipMissingProperties(true)
     val constructor = new Constructor(classOf[ClusterProperties], new LoaderOptions())
     val yamlObjNested = new Yaml(constructor, representer)
-    Option(yamlObjNested.load(clusterProps).asInstanceOf[ClusterProperties])
+    val loadedClusterProps = yamlObjNested.load(clusterProps).asInstanceOf[ClusterProperties]
+    if (loadedClusterProps != null && loadedClusterProps.softwareProperties == null) {
+      loadedClusterProps.softwareProperties = new util.LinkedHashMap[String, String]()
```

Comment:
Would it better to add log message that softwareProperties is empty?

Reply:
Thanks! Changes updated.